### PR TITLE
Feature/aql subquery execution block impl execute implementation exec block impl tests shadow rows

### PR DESCRIPTION
**Pull request by mchacki**
Internal PR.

This adds tests for ShadowRows and fixes small issues within the bahviour of ExecutionBlockImpl found during the test.

Also adds a shouldSkip() helper method on AqlCall.

Jenkins:
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/8250/

Should be merged after:
https://github.com/arangodb/arangodb/pull/10838

as they overlap

### CODE REVIEW

**Comment by mchacki** [comment]
The above Jenkins was all good.

As we have merged in devel, I have added a new run (just to be sure):
No merge conflicts encountered.

http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/8254/

**Comment by mchacki** [comment]
The above jenkins did trigger 3 of the known issues, they are not in any way related to this PR.

File: tests/Aql/ExecutionBlockImplTest.cpp [hunk 1]
```diff
@@ -1494,6 +1824,477 @@ TEST_P(ExecutionBlockImplExecuteIntegrationTest, test_call_forwarding_implement_
...
+    ValidateResult(builder, skipped, block, outReg, 2);
+    ASSERT_TRUE(block != nullptr);
+    // Include both shadow rows
+    ValidateShadowRow(block, block->size() - 2, 0);
+    ValidateShadowRow(block, block->size() - 1, 1);
+  }
+}
+
+// Test multiple subqueries
+TEST_P(ExecutionBlockImplExecuteIntegrationTest, multiple_subqueries) {
```
Comment (markuspf) [inline tests/Aql/ExecutionBlockImplTest.cpp]: After the experience with bugs pertaining to empty inputs and nested subqueries, we should explicitly test empty inputs and nested subqueries, I think.

File: tests/Aql/WaitingExecutionBlockMock.h [hunk 2]
```diff
@@ -45,16 +45,27 @@ namespace aql {
  */
 class WaitingExecutionBlockMock final : public arangodb::aql::ExecutionBlock {
  public:
+  /**
+   * @brief Define how often this Block should return "WAITING"
+   */
+  enum WaitingBehaviour {
+    NEVER,  // Never return WAITING
+    ONCE,  // Return WAITING on the first execute call, afterwards return all blocks
+    ALLWAYS  // Return Waiting once for every execute Call.
```
Comment (markuspf) [inline tests/Aql/WaitingExecutionBlockMock.h]:
```suggestion
    ALWAYS  // Return Waiting once for every execute Call.
```

Comment (markuspf) [reply tests/Aql/WaitingExecutionBlockMock.h]: This has to be applied throughout of course

Comment (mchacki) [reply tests/Aql/WaitingExecutionBlockMock.h]: I guess I will never learn to type this correct...

**Comment by mchacki** [comment]
After addressing the comments.

New Jenkins:
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/8314/

**Comment by mchacki** [comment]
Jenkins okay.

**Pull request merged**